internal/bundle/plain.go

```diff
@@ -0,0 +1,57 @@
+package bundle
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PlainV0 struct {
+	Objects []client.Object
+}
+
+func LoadPlainV0(bundle fs.FS) (PlainV0, error) {
+	const manifestsDir = "manifests"
+	var objects []client.Object
+	dirEntries, err := fs.ReadDir(bundle, manifestsDir)
+	if err != nil {
+		return PlainV0{}, err
+	}
+	if len(dirEntries) == 0 {
+		return PlainV0{}, errors.New("invalid bundle: found zero files")
+	}
+	for _, manifest := range dirEntries {
+		if manifest.IsDir() {
+			return PlainV0{}, fmt.Errorf("subdirectories are not allowed within the %q directory "+
+				"of the bundle image filesystem: found %q", manifestsDir, filepath.Join(manifestsDir, manifest.Name()))
+		}
+		manifestReader, err := bundle.Open(filepath.Join("manifests", manifest.Name()))
+		if err != nil {
+			return PlainV0{}, err
+		}
+		defer manifestReader.Close()
+		// Stream().Do() visits and serializes each object
+		result := resource.NewLocalBuilder().Flatten().Unstructured().Stream(manifestReader, "Load PlainV0 Bundle").Do()
```

Comment:
Rather than using `"Load PlainV0 Bundle"`, I think the intention of that string is to be something descriptive about the `manifestReader`. I'd suggest extracting `filepath.Join("manifests", manifest.Name())` as a variable and using that when we `Open` and here.

Reply:
Looks like this one got lost in the latest push?

Reply:
nice catch thanks